Sources/SMTPKitten/SMTPMessage.swift

```diff
@@ -43,3 +43,28 @@ public enum SMTPResponseCode: Int {
     case startMailInput = 354
     case commandNotRecognized = 502
 }
+
+/// SMTP Authentication method.
+public enum SMTPAuthMethod: String, CaseIterable {
+    case plain = "PLAIN"
+    case login = "LOGIN"
+    case crammd5 = "CRAM-MD5"
+    
+    var priority: Int {
+        switch self {
+        case .plain: 0
+        case .login: 1
+        case .crammd5: 2
+        }
+    }
+}
+
+/// Type for building a correct credentials for PLAIN SMTP auth.
+public struct SMTPPlainCreds: SMTPClientRequest {
+    public let user: String
+    public let password: String
+    
+    public var text: String {
+        "\0\(user)\0\(password)"
+    }
+}
```

Comment:
Do we need this to be public?

```suggestion
/// Type for building a correct credentials for PLAIN SMTP auth.
struct SMTPPlainCreds: SMTPClientRequest {
    let user: String
    let password: String
    
    var text: String {
        "\0\(user)\0\(password)"
    }
}
```